src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -295,8 +309,48 @@ object ArrayBuffer extends StrictOptimizedSeqFactory[ArrayBuffer] {
   }
 }
 
-final class ArrayBufferView[A](val array: Array[AnyRef], val length: Int) extends AbstractIndexedSeqView[A] {
+// TODO: use `CheckedIndexedSeqView.Id` once we can change the return type of
+final class ArrayBufferView[A] private[mutable](underlying: ArrayBuffer[A], mutationCount: () => Int)
```

Comment:
What is the benefit of passing `mutationCount` as a separate parameter, rather than making it `private[mutable]` in class `ArrayBuffer`?

Reply:
none that I can think of. I think the reason I did it this way was just to mirror the regular `CheckedIndexedSeqView`, which is what we'll hopefully use eventually. I don't really have strong preferences

Reply:
I'm going to hit "resolve" here since nobody has expressed a strong opinion either way.